src/core/Akka/Actor/ActorRef.Extensions.cs

```diff
@@ -23,6 +25,17 @@ public static bool IsNobody(this IActorRef actorRef)
         {
             return actorRef == null || actorRef is Nobody || actorRef is DeadLetterActorRef;
         }
+
+        /// <summary>
+        /// Returns the <paramref name="actorRef"/>'s value if it's not <see langword="null"/>, <see cref="Nobody"/>, 
+        /// or <see cref="DeadLetterActorRef"/>. Otherwise return the result of evaluating `elseValue`.
+        /// </summary>
+        /// <param name="actorRef">The actor that is being tested.</param>
+        /// <param name="elseValue">TBD</param>
+        public static IActorRef GetOrElse(this IActorRef actorRef, Func<IActorRef> elseValue)
+        {
+            return actorRef.IsNobody() ? elseValue() : actorRef;
```

Comment:
Akka actually checks if it's not `isEmpty`, but I'm not sure if we should check `actorRef.IsNobody()`, `actorRef == null || actorRef is Nobody`, or just `actorRef == null`.

Reply:
`IsNobody` performs all of those checks internally - does a null check, `is Nobody`, and also checks if the ref is a `DeadLetterActorRef`

Reply:
Sure, what I meant was that in the JVM they just check for isEmpty (which I think is just NULL).